crypto/bn/rsaz_exp.c

```diff
@@ -309,6 +311,8 @@ void RSAZ_512_mod_exp(BN_ULONG result[8],
     /* from Montgomery */
     rsaz_512_mul_by_one(result, temp, m, k0);
 
+    bn_reduce_once_in_place(result, /*carry=*/0, m, storage, 8);
```

Comment:
BoringSSL doesn't have this code---we don't consider RSA-1024 worth optimizing for these days---so I don't know off-hand whether it is correct. But supposing it's the same algorithm as rsaz_1024 (RSA-2048), the fix looks right.